rust/protocol/src/crypto.rs

```diff
@@ -13,25 +17,31 @@ use hmac::{Hmac, Mac, NewMac};
 use sha2::Sha256;
 use subtle::ConstantTimeEq;
 
-pub fn aes_256_ctr_encrypt(ptext: &[u8], key: &[u8]) -> Result<Vec<u8>> {
-    if key.len() != 32 {
-        return Err(SignalProtocolError::InvalidCipherCryptographicParameters(
-            32, 0,
-        ));
-    }
+const BYTE: usize = 8;
```

Comment:
I'd call this `BITS_PER_BYTE` if you think it's useful to specify the sizes in bits.

Reply:
I like specifying them in bits so that they can be used to index into a slice directly without confusion about whether you should multiply by 8 yourself. I think the nicer solution to this would be a library that uses types for units to avoid ever getting this wrong.